values.yaml

```diff
@@ -766,46 +766,53 @@ meshGateway:
+    #   If service.type=LoadBalancer use the external IP or hostname of
+    #   the service. Use the port set by service.port.
+    #   If service.type=NodePort use the Node IP. The port will be set to
+    #   service.nodePort so this cannot be null.
+    #   If service.type=ClusterIP use the ClusterIP. The port will be set to
+    #   service.port.
+    #   service.type=ExternalName is not supported.
+    # NodeIP - The node ip as provided by the Kubernetes downward API.
+    # NodeName - The name of the node as provided by the Kubernetes downward
+    #   API. This is useful if the node names are DNS entries that
+    #   are routable from other datacenters.
+    # Static - Use the address hardcoded in meshGateway.wanAddress.static.
+    source: "Service"
+
+    # Port that gets registered for WAN traffic.
+    # If source is set to "Service" then this setting will have no effect.
+    # See the documentation for source as to which port will be used in that
+    # case.
     port: 443
 
-    # If true, each Gateway Pod will advertise its NodeIP
-    # (as provided by the Kubernetes downward API) as the wan address.
-    # This is useful if the node IPs are routable from other DCs.
-    # useNodeName and host must be false and "" respectively.
-    useNodeIP: true
-
-    # If true, each Gateway Pod will advertise its NodeName
-    # (as provided by the Kubernetes downward API) as the wan address.
-    # This is useful if the node names are DNS entries that are
-    # routable from other DCs.
-    # meshGateway.wanAddress.port will be used as the port for the wan address.
-    # useNodeIP and host must be false and "" respectively.
-    useNodeName: false
-
-    # If set, each gateway Pod will use this host as its wan address.
-    # Users must ensure that this address routes to the Gateway pods,
-    # for example via a DNS entry that routes to the Service fronting the Deployment.
-    # meshGateway.wanAddress.port will be used as the port for the wan address.
-    # useNodeIP and useNodeName must be false.
-    host: ""
+    # If source is set to "Static" then this value will be used as the wan
+    # address of the mesh gateways. This is useful if you've configured a
+    # DNS entry to point to your mesh gateways.
+    static: ""
 
   # The service option configures the Service that fronts the Gateway Deployment.
   service:
     # Whether to create a Service or not.
-    enabled: false
+    enabled: true
 
     # Type of service, ex. LoadBalancer, ClusterIP.
-    type: ClusterIP
+    type: LoadBalancer
 
     # Port that the service will be exposed on.
     # The targetPort will be set to meshGateway.containerPort.
     port: 443
 
-    # Optional nodePort of the service. Can be used in conjunction with
-    # type: NodePort.
+    # Optionally hardcode the nodePort of the service if using type: NodePort.
+    # If not set and using type: NodePort, Kubernetes will automatically assign
+    # a port.
```

Comment:
Right now, the helm chart fails if I don't have this set:
```
Error: template: consul/templates/mesh-gateway-deployment.yaml:141:69: executing "consul/templates/mesh-gateway-deployment.yaml" at <fail "if meshGateway.wanAddress.source=Service and meshGateway.service.type=NodePort, meshGateway.service.nodePort must be set">: error calling fail: if meshGateway.wanAddress.source=Service and meshGateway.service.type=NodePort, meshGateway.service.nodePort must be set
```

Reply:
Yes because we have to set the port in the addresses section of the service definition so we need to know what it is ahead of time

Reply:
Sure, I understand. I think we need to say that we require it in the comment. But maybe, since Kube doesn't require it, we could get it by running the `service-address` command?

Reply:
It would need to be a separate command/flag since it needs to be templated separately into the service definition.

After talking we decided to leave it required for now and we can later make it optional.